drivers/sdhc/sdhci_common.c

```diff
@@ -695,14 +809,32 @@ static int sdhci_check_version(struct sdhci_common *sdhci_ctx)
 	return 0;
 }
 
-int sdhci_enable_dma(struct sdhci_common *sdhci_ctx, enum sdhc_dma_select dma_mode)
+int sdhci_enable_sdma(struct sdhci_common *sdhci_ctx)
 {
-	if (dma_mode != SDHC_DMA_SDMA) {
-		return -ENOTSUP;
+	sdhci_ctx->f_use_dma = true;
+	sdhci_ctx->dma_mode = SDHC_DMA_SDMA;
+
+	return 0;
+}
+
+int sdhci_enable_adma2(struct sdhci_common *sdhci_ctx, struct sdhc_adma_desc *descs_table,
+		       uint32_t descs_table_size)
+{
+	if (!descs_table || !descs_table_size) {
+		return -EINVAL;
 	}
 
+#if defined(CONFIG_64BIT)
```

Comment:
I don't know exactly how it works, but as far as I understand, it is possible to have addresses bigger than 32 bits on a 32-bit system. This is known as PAE for x86 and LPAE for ARM.

Reply:
Those >32bit addrs usually are not supported for DMA - at least I'm not aware of  such cases. In such, systems DMA is allowed only in <32bit area.